timesketch/frontend-ng/src/components/LeftPanel/AnalyzerResults.vue

```diff
@@ -236,32 +238,90 @@ export default {
         }
 
         if (
           perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].last_analysissession_id <
           analysis.analysissession_id
         ) {
-          // this timeline is already in the results for this analyzer but check if the session is newer and update it
-          perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].created_at = analysis.created_at
-          perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].verdict = analysis.result
-          perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].last_analysissession_id =
-            analysis.analysissession_id
-          perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].analysis_status =
-            analysis.status[0].status
+          // there is a newer session for this analyzer on this timeline. Update the results.
+          if (perAnalyzer[analysis.analyzer_name].analyzerInfo.is_multi) {
+            multiResults = this.getAnalyzerMultiResults(analyses, analysis.analyzer_name, analysis.analysissession_id, analysis.timeline.id)
+            perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].last_analysissession_id =
+              analysis.analysissession_id
+            perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].results[0].analysis_status = multiResults.analysis_status
+            perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].results = multiResults.results
+          } else {
+            perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].results[0].created_at = analysis.created_at
+            perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].results[0].verdict = analysis.result
+            perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].last_analysissession_id =
+              analysis.analysissession_id
+            perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].analysis_status = analysis.status[0].status
+            perAnalyzer[analysis.analyzer_name].timelines[analysis.timeline.name].results[0].analysis_status =
+              analysis.status[0].status
+          }
         }
       }
       return perAnalyzer
     },
+    getAnalyzerMultiResults(analyses, analyzerName, sessionId, timelineId) {
+      let results = []
+      let status = new Set()
+      for (const analysis of analyses) {
+        if (analysis.analyzer_name === analyzerName && analysis.analysissession_id === sessionId && analysis.timeline.id === timelineId) {
+          results.push({
+            verdict: analysis.result,
+            created_at: analysis.created_at,
+            analysis_status: analysis.status[0].status,
+          })
+          status.add(analysis.status[0].status)
+        }
+      }
+      if (status.size === 1 && (status.has('DONE') || status.has('ERROR'))) {
+        return {
+          results: results,
```

Comment:
nit: use the short form
```javascript
 return {
  results,
  analysis_status: [...status][0],
}
```

The same for the other occurrences of `results: results`.

As an alternative you could also pull out the logic into a method that takes `status` and returns an `analysis_status` and use it like:

```javascript
return {
  results,
  analysis_status: this.analysisStatus(status),
}
```